src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -297,6 +298,111 @@ public AgentSettings LoadSettings()
             }
         }
 
+        public async Task UnconfigureAsync(CommandSettings command)
+        {
+            //stop, uninstall service and remove service config file
+            _term.Write(StringUtil.Loc("UninstallingService"));
+            if (_store.IsServiceConfigured())
+            {
+                var serviceControlManager = HostContext.GetService<IServiceControlManager>();
+                try
+                {
+                    serviceControlManager.UnconfigureService();
+                    _term.WriteLine(StringUtil.Loc("Success"));
+                }
+                catch (Exception ex)
+                {
+                    Trace.Error(ex);
+                    _term.WriteLine(StringUtil.Loc("Failed"));
+                    throw;
+                }
+            }
+            else
+            {
+                _term.WriteLine(StringUtil.Loc("NotFound"));
+            }
+
+            //delete agent from the server
+            _term.Write(StringUtil.Loc("UnregisteringAgent"));
+            bool isConfigured = _store.IsConfigured();
+            bool hasCredentials = _store.HasCredentials();
+            if (isConfigured && hasCredentials)
```

Comment:
the "if (isConfigured && hasCredentials)" check is needed, because if don't have config files, we cannot connect to the server and thus cannot delete the agent registration.